res/config/game_script/timetable_gui.lua

```diff
@@ -1,1269 +1,1196 @@
-}
-local co = nil
-local state = nil
-
-local timetableChanged = false
-
-local stationTableScrollOffset
-local lineTableScrollOffset
-local constraintTableScrollOffset
-
-
-local UIStrings = {
-        arr	= _("arr_i18n"),
-		arrival	= _("arrival_i18n"),
-		dep	= _("dep_i18n"),
-		departure = _("departure_i18n"),
-		unbunch_time = _("unbunch_time_i18n"),
-		unbunch	= _("unbunch_i18n"),
-        auto_unbunch = _("auto_unbunch_i18n"),
-		timetable = _("timetable_i18n"),
-		timetables = _("timetables_i18n"),
-		line = _("line_i18n"),
-		lines = _("lines_i18n"),
-		min	= _("time_min_i18n"),
-		sec	= _("time_sec_i18n"),
-		stations = _("stations_i18n"),
-		frequency = _("frequency_i18n"),
-		journey_time = _("journey_time_i18n"),
-		arr_dep	= _("arr_dep_i18n"),
-		no_timetable = _("no_timetable_i18n"),
-		all	= _("all_i18n"),
-		add	= _("add_i18n"),
-		none = _("none_i18n"),
-		tooltip	= _("tooltip_i18n")
-}
-
-local local_styles = {
-    zh_CN = "timetable-mono-sc",
-    zh_TW = "timetable-mono-tc",
-    ja = "timetable-mono-ja",
-    kr = "timetable-mono-kr"
-}
-
--------------------------------------------------------------
----------------------- stationTab ---------------------------
--------------------------------------------------------------
--- abbreviated prefix: st
-
-function timetableGUI.initStationTab()
-    if menu.stationTabScrollArea then UIState.floatingLayoutStationTab:removeItem(menu.scrollArea) end
-
-    --left table
-    local stationOverview = api.gui.comp.TextView.new('StationOverview')
-    menu.stationTabScrollArea = api.gui.comp.ScrollArea.new(stationOverview, "timetable.stationTabStationOverviewScrollArea")
-    menu.stStations = api.gui.comp.Table.new(1, 'SINGLE')
-    menu.stationTabScrollArea:setMinimumSize(api.gui.util.Size.new(320, 720))
-    menu.stationTabScrollArea:setMaximumSize(api.gui.util.Size.new(320, 720))
-    menu.stationTabScrollArea:setContent(menu.stStations)
-    timetableGUI.stFillStations()
-    UIState.floatingLayoutStationTab:addItem(menu.stationTabScrollArea,0,0)
-
-    menu.stationTabLinesScrollArea = api.gui.comp.ScrollArea.new(api.gui.comp.TextView.new('LineOverview'), "timetable.stationTabLinesScrollArea")
-    menu.stationTabLinesTable = api.gui.comp.Table.new(1, 'NONE')
-    menu.stationTabLinesScrollArea:setMinimumSize(api.gui.util.Size.new(880, 720))
-    menu.stationTabLinesScrollArea:setMaximumSize(api.gui.util.Size.new(880, 720))
-    -- menu.stationTabLinesTable:setColWidth(0,23)
-    -- menu.stationTabLinesTable:setColWidth(1,150)
-
-    menu.stationTabLinesScrollArea:setContent(menu.stationTabLinesTable)
-    UIState.floatingLayoutStationTab:addItem(menu.stationTabLinesScrollArea,1,0)
-end
-
--- fills the station table on the left side with all stations that have constraints
-function timetableGUI.stFillStations()
-    -- list all stations that are part of a timetable 
-    timetable.cleanTimetable() -- remove old lines no longer in the game
-    timetableChanged = true
```

Comment:
Why is `timetableChanged` being set to `false`?

Reply:
Because this function only fills the station list in the GUI without changing the content of the timetable. This way I prevent the timetable object from being created again when the station list is displayed.

Reply:
Ok I agree. Should this line be omitted rather than set to false? Something else might change at the same time that does cause the timetable to change.

Reply:
> Ok I agree. Should this line be omitted rather than set to false? Something else might change at the same time that does cause the timetable to change.

To remove it would maybe be better. Your right.